File: mobile/lib/models/wallet.dart
class Wallet {
  final String id;
  final String tripId;
  final String currency;
  final int balancePaise;
  final String status;

  Wallet({
    required this.id,
    required this.tripId,
    required this.currency,
    required this.balancePaise,
    required this.status,
  });

  factory Wallet.fromJson(Map<String, dynamic> json) {
    return Wallet(
      id: json['id'],
      tripId: json['trip_id'],
      currency: json['currency'] ?? 'INR',
      balancePaise: json['balance_paise'] ?? 0,
      status: json['status'] ?? 'ACTIVE',
    );
  }

  double get balance => balancePaise / 100;
}

File: backend/app/models/wallet.py
import uuid

from datetime import datetime

from sqlalchemy import (
    BigInteger,
    DateTime,
    ForeignKey,
    String,
)
from sqlalchemy.orm import Mapped, mapped_column

from app.db.database import Base


class Wallet(Base):

    __tablename__ = "wallets"

    id: Mapped[uuid.UUID] = mapped_column(
        primary_key=True,
        default=uuid.uuid4,
    )

    trip_id: Mapped[uuid.UUID] = mapped_column(
        ForeignKey("trips.id", ondelete="CASCADE"),
        unique=True,
        nullable=False,
    )

    currency: Mapped[str] = mapped_column(
        String(3),
        default="INR",
        nullable=False,
    )

    balance_paise: Mapped[int] = mapped_column(
        BigInteger,
        default=0,
        nullable=False,
    )

    status: Mapped[str] = mapped_column(
        String(20),
        default="ACTIVE",
        nullable=False,
    )

    created_at: Mapped[datetime] = mapped_column(
        DateTime,
        default=datetime.utcnow,
        nullable=False,
    )

    updated_at: Mapped[datetime] = mapped_column(
        DateTime,
        default=datetime.utcnow,
        onupdate=datetime.utcnow,
        nullable=False,
    )